### DISCUSSION TITLE
Disallowing creation of new topics

### DISCUSSION THREAD

**Discussion by mckravchyk** [giscus/giscus#1302] [General]
The guide recommends using a repository of Announcement type so only "you and Giscus can create new topics". Am I missing something? I think it's somewhat misleading because there's nothing stopping someone from using the snippet on any Giscus-enabled repository that they do not own to create new issues. Or did Giscus created the topic automatically only because I commented myself with the owner account?

Is there a way to disable automatic topic creation on the back-end side of it?

**Comment by laymonage** [comment]
> I think it's somewhat misleading because there's nothing stopping someone from using the snippet on any Giscus-enabled repository that they do not own to create new issues.

Yes, but the point of the automatic discussion creation feature is so that you don't have to create the discussion for every page, and giscus will do so when anyone leaves the first comment. Someone using the snippet targeting another repository is just a more elaborate way to do the same thing.

> Is there a way to disable automatic topic creation on the back-end side of it?

The `origins` specifier in `giscus.json` on your repo can help prevent the widget from loading your repository on other websites: https://github.com/giscus/giscus/blob/main/ADVANCED-USAGE.md#origins

This does not completely solve the issue though, as the API will still be exposed to it. I suppose we can add another setting to `giscus.json` to completely disable discussion creation in the API.

**Comment by mckravchyk** [comment]
> The origins specifier in giscus.json on your repo can help prevent the widget from loading your repository on other websites

That's great, thanks!

> I suppose we can add another setting to giscus.json to completely disable discussion creation in the API.

I guess that would be nice to have - then the front-end would need to indicate that the discussion has to be created manually when it's not there instead of the comment box.

Thanks for your reply and making this app - it is really cool!

**Reply by laymonage** [reply]
> then the front-end would need to indicate that the discussion has to be created manually when it's not there

If you don't mind specifying the discussion number for each page where you want comments enabled, I'd suggest using the `number` mapping, so giscus will only use existing discussions.

**Reply by mckravchyk** [reply]
Yeah, that sounds like a better workflow :)
